### PR TITLE
Add "Send to Sailthru" Product Grid Mass Action

### PR DESCRIPTION
**Pull request by silverman63**
This adds a new *BETA* feature to the integration, allowing sync'ing of up to 350 products at a time to Sailthru via Mass Action. The mass action will be visible the admin product catalog view when in single-store mode or when at store scope. 

image

### CODE REVIEW

File: app/code/community/Sailthru/Email/Model/Observer/Adminhtml.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    /**
+     * Catch loading of Catalog Product Admin and insert a new mass action to send to Sailthru.
+     * @param Varien_Event_Observer $observer
+     */
+    public function addBlockMassAction(Varien_Event_Observer $observer) {
+        $block = $observer->getEvent()->getBlock();
+
+        // verify this is for a specific store, or a single-store Magento instance
+        $store = $block->getRequest()->getParam('store');
+        if (!$store and count(Mage::app()->getStores()) == 1) {
```
Comment (nickgundry) [inline app/code/community/Sailthru/Email/Model/Observer/Adminhtml.php]: Is this the store Id? Is it always 1 if there is only one store.

File: app/code/community/Sailthru/Email/controllers/ContentController.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+     * <magento_uri>/content/bulk endpoint action
+     */
+    public function bulkAction() {
+        $data = $this->getRequest()->getPost();
+        if ($data and array_key_exists('product', $data) and
+            array_key_exists('massaction_prepare_key', $data) and $data['massaction_prepare_key'] == 'product' and
+            array_key_exists('store', $data) and intval($data['store']) > 0) {
+            $this->_processItems($data['product'], $data['store']);
+        } else {
+            Mage::getSingleton('adminhtml/session')->addError("Request to bulk update Sailthru was malformed. Please try again.");
```
Comment (nickgundry) [inline app/code/community/Sailthru/Email/controllers/ContentController.php]: "was malformed" doesn't give them much info, would "Request to bulk update Sailthru failed" be more appropriate? What would they do to fix it?

File: app/code/community/Sailthru/Email/Model/Observer/Adminhtml.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            $store = 1;
+        }
+
+        // we only want to add this to the Product Catalog mass action block
+        if($store != null and get_class($block) =='Mage_Adminhtml_Block_Widget_Grid_Massaction'
+            && $block->getRequest()->getControllerName() == 'catalog_product') {
+
+            /** @var Mage_Adminhtml_Block_Widget_Grid_Massaction_Abstract $block */
+            $block->addItem('sailthruemail_content_bulk', array(
+                'label' => 'Send to Sailthru Content Library',
```
Comment (nickgundry) [inline app/code/community/Sailthru/Email/Model/Observer/Adminhtml.php]: Is this the line that defines the drop down? 
Let's do [BETA] Send to Sailthru like you suggested if so.

**Review by nickgundry — CHANGES_REQUESTED** [review]

File: app/code/community/Sailthru/Email/Model/Observer/Adminhtml.php (hunk 1 above)
Comment (silverman63) [reply app/code/community/Sailthru/Email/Model/Observer/Adminhtml.php]: The request param can be null if out of store scope, or -I believe- in single-store mode.

File: app/code/community/Sailthru/Email/controllers/ContentController.php (hunk 2 above)
Comment (silverman63) [reply app/code/community/Sailthru/Email/controllers/ContentController.php]: No idea why that error would occur. Shouldn't happen. Hence debug POST info available.

File: app/code/community/Sailthru/Email/Model/Observer/Adminhtml.php (hunk 3 above)
Comment (silverman63) [reply app/code/community/Sailthru/Email/Model/Observer/Adminhtml.php]: This was updated. Believe you were looking at an old commit.

File: app/code/community/Sailthru/Email/Model/Observer/Adminhtml.php [hunk 4]
```diff
@@ -23,7 +23,7 @@ public function addBlockMassAction(Varien_Event_Observer $observer) {
 
             /** @var Mage_Adminhtml_Block_Widget_Grid_Massaction_Abstract $block */
             $block->addItem('sailthruemail_content_bulk', array(
-                'label' => 'Send to Sailthru Content Library',
+                'label' => '[BETA] Send to Sailthru',
```
Comment (nickgundry) [inline app/code/community/Sailthru/Email/Model/Observer/Adminhtml.php]: +1

**Review by nickgundry — APPROVED** [review]

**Pull request merged**